### PR TITLE
Updated trait attributes so they can be applied at class level

### PR DESCRIPTION
**Pull request by devr24**
Making testing more flexible by allowing our xUnit trait filter to be applied at Class level as well as Method level

### CODE REVIEW

**Review by dguerin — APPROVED** [review]

**Comment by dguerin** [comment]
@rmccabe24 and the version number too.

**Review by tabman83 — COMMENTED** [review]
You need to update the package version number and release comments otherwise we won't be able to release the package on nuget.

**Comment by VaclavK** [comment]
I suspect this was kinda intentional per method

if you do it at class level it will kinda lead to people creating separate classes and perhaps even projects for unit vs int vs whatever which really was not the original "vision"

technically no issue though

**Comment by devr24** [comment]
How do I update the release comments guys? @tabman83 @VaclavK 

@VaclavK - I understand what you mean, although from the usage I've seen, people typically do seem to have separate unit and integration test projects.  Personally, when writing tests I do have integration tests in one class and unit in another, therefore having at class level is much more convenient than for each method.

File: src/Eshopworld.Tests.Core/Eshopworld.Tests.Core.csproj [hunk 1]
```diff
@@ -4,7 +4,7 @@
     <TargetFramework>netstandard2.1</TargetFramework>
     <IsPackable>true</IsPackable>
 
-    <Version>2.0.3</Version>
+    <Version>2.0.4</Version>
     <PackageReleaseNotes>Upgrade to netstandard2.1</PackageReleaseNotes>
```
Comment (tabman83) [inline src/Eshopworld.Tests.Core/Eshopworld.Tests.Core.csproj]: Can you please add a description of what this change is about?

**Review by tabman83 — CHANGES_REQUESTED** [review]

File: src/Eshopworld.Tests.Core/Eshopworld.Tests.Core.csproj (hunk 1 above)
Comment (tabman83) [reply src/Eshopworld.Tests.Core/Eshopworld.Tests.Core.csproj]: Inside the `<PackageReleaseNotes>` tag! Cheers!

Comment (devr24) [reply src/Eshopworld.Tests.Core/Eshopworld.Tests.Core.csproj]: Just the `<PackageReleaseNotes>` tag? - sure

**Comment by VaclavK** [comment]
> 
> 
> How do I update the release comments guys? @tabman83 @VaclavK
> 
> @VaclavK - I understand what you mean, although from the usage I've seen, people typically do seem to have separate unit and integration test projects. Personally, when writing tests I do have integration tests in one class and unit in another, therefore having at class level is much more convenient than for each method.

people do all kinds of things :D

I used to do that too but stopped - after seeing some of the tooling work - because that sepaation is quite artificial and means I have to do lot more clicks.. the traits express what it is so separating by class/project even is then duplication and ultimately muddles things

my 2c... if people want to do separate projects/classes and the pain with it.. sure.. .I can keep dping it per method so no biggie

File: src/Eshopworld.Tests.Core/Eshopworld.Tests.Core.csproj (hunk 1 above)
Comment (tabman83) [reply src/Eshopworld.Tests.Core/Eshopworld.Tests.Core.csproj]: Brilliant thanks

**Review by tabman83 — APPROVED** [review]

**Comment by devr24** [comment]
Thanks guys!

**Comment by dguerin** [comment]
> > How do I update the release comments guys? @tabman83 @VaclavK
> > @VaclavK - I understand what you mean, although from the usage I've seen, people typically do seem to have separate unit and integration test projects. Personally, when writing tests I do have integration tests in one class and unit in another, therefore having at class level is much more convenient than for each method.
> 
> people do all kinds of things :D
> 
> I used to do that too but stopped - after seeing some of the tooling work - because that sepaation is quite artificial and means I have to do lot more clicks.. the traits express what it is so separating by class/project even is then duplication and ultimately muddles things
> 
> my 2c... if people want to do separate projects/classes and the pain with it.. sure.. .I can keep dping it per method so no biggie

Yeah I agree but I think teams should be given options and in some cases it will be at the method level and others at the class level. That being said last I checked there is a unit and integration test project in the template so it would appear that the separation is encourage at the moment.

That all all being said my personal preference is to have it at the method level for the very same reasons Oisin describes. Thus allowing a single class to have tests for a unit or units of code which are tested at different levels of granularity.

**Pull request merged**